x/staking/types/msg.go

```diff
@@ -24,6 +24,13 @@ const (
 	TypeMsgCreateValidator = "create_validator"
 	TypeMsgDelegate        = "delegate"
 	TypeMsgBeginRedelegate = "begin_redelegate"
+
+	// These are used for querying events by action.
+	TypeSvcMsgUndelegate      = "/cosmos.staking.v1beta1.Msg/Undelegate"
```

Comment:
Are these FQ paths not defined elsewhere already?

Reply:
Unfortunately not... In Aaron's custom code generator they are, and i think that's useful.